src/arch/src/aarch64/fdt.rs

```diff
@@ -129,12 +129,9 @@ pub fn create_fdt<T: DeviceInfoForFDT + Clone + Debug>(
 
     // Write FDT to memory.
     let fdt_address = GuestAddress(get_fdt_addr(&guest_mem));
-    let written = guest_mem
+    guest_mem
         .write_slice(fdt_final.as_slice(), fdt_address)
         .map_err(Error::WriteFDTToMemory)?;
-    if written < FDT_MAX_SIZE {
-        return Err(Error::IncompleteFDTMemoryWrite);
```

Comment:
I removed `IncompleteFDTMemoryWrite`.

I don't think such a method exists. At least not that I know of. And for the moment I don't know of any scenario in which we would need to retry. 

Probably the idea is that  `write_slice` doesn't need to return the number of written bytes. The number of written bytes is `buf.len()` in case of success.

Reply:
Yeah, I just realized afterwards that you are suppose to get the number of written bytes from the error, which I find a bit weird tbh.

Reply:
If it's expected to read only a part of the slice, it is also possible to use `mem.read_from` since slices implement `Read`. Usually it's not expected (as is this case) and vm-memory's semantics allows using `?` more effectively.